modules/indexing/src/main/java/org/apache/ignite/internal/mxbean/SqlQueryMXBean.java

```diff
@@ -178,6 +178,14 @@
         "heavy analytics query.")
     String getSqlQueryMemoryQuota();
 
+    /**
+     * Gets the amount of memory available for SQL queries. This is equal to the {@link #getSqlGlobalMemoryQuotaBytes()} minus the amount of memory currently occupied by the queries.
+     *
+     * @return Amount of memory bytes available for SQL queries.
+     */
+    @MXBeanDescription("The amount of memory available for SQL queries. This is equal to SqlGlobalMemoryQuotaBytes minus the amount of memory currently occupied by the queries.")
+    Long getSqlFreeMemoryBytes();
```

Comment:
Why not `long`? What for the boxing here? Moreover `QueryMemoryManager#freeMemory` returns just `long`